File: Android/data/bet/src/main/java/com/felipearpa/tyche/data/bet/application/GetLivePoolGamblerBets.kt
package com.felipearpa.tyche.data.bet.application

import com.felipearpa.tyche.data.bet.domain.PoolGamblerBetRepository

class GetLivePoolGamblerBets(private val poolGamblerBetRepository: PoolGamblerBetRepository) {
    suspend fun execute(
        poolId: String,
        gamblerId: String,
        next: String? = null,
        searchText: String? = null,
    ) =
        poolGamblerBetRepository.getLivePoolGamblerBets(
            poolId = poolId,
            gamblerId = gamblerId,
            next = next,
            searchText = searchText,
        )
}

File: Amazon.Lambda/Felipearpa.Tyche.Pool/src/Application/GetLivePoolGamblerBets.fs
namespace Felipearpa.Tyche.Pool.Application

open Felipearpa.Tyche.Pool.Domain
open Felipearpa.Type

type GetLivePoolGamblerBets(poolGamblerBetRepository: IPoolGamblerBetRepository) =

    member this.ExecuteAsync(poolId: Ulid, gamblerId: Ulid, searchText: string option, next: string option) =
        poolGamblerBetRepository.GetLivePoolGamblerBetsAsync(poolId, gamblerId, searchText, next)
